gapic-generator/templates/default/helpers/presenters/service_presenter.rb

```diff
@@ -54,85 +59,78 @@ def name
     @service.name
   end
 
-  def service_proto_name_full
-    ruby_namespace @service.address
+  def proto_service_name_full
+    "#{namespace}::#{name}"
   end
 
-  def service_proto_file_path
+  def proto_service_file_path
     if @service.parent
       @service.parent.name.sub ".proto", "_pb.rb"
     else
       ruby_file_path(@service).sub ".rb", "_pb.rb"
     end
   end
 
-  def service_proto_file_name
-    service_proto_file_path.split("/").last
+  def proto_service_file_name
+    proto_service_file_path.split("/").last
   end
 
-  def service_proto_require
-    service_proto_file_path.sub ".rb", ""
-  end
-
-
-  def services_proto_name_full
-    ruby_namespace @service.address
+  def proto_service_require
+    proto_service_file_path.sub ".rb", ""
   end
 
-  def services_proto_file_path
+  def proto_services_file_path
     if @service.parent
       @service.parent.name.sub ".proto", "_services_pb.rb"
     else
       ruby_file_path(@service).sub ".rb", "_services_pb.rb"
     end
   end
 
-  def services_proto_file_name
-    services_proto_file_path.split("/").last
+  def proto_services_file_name
+    proto_services_file_path.split("/").last
   end
 
-  def services_proto_require
-    services_proto_file_path.sub ".rb", ""
+  def proto_services_require
+    proto_services_file_path.sub ".rb", ""
   end
 
-  def services_stub_name_full
-    "#{services_proto_name_full}::Stub"
+  def proto_service_stub_name_full
+    "#{proto_service_name_full}::Stub"
```

Comment:
Yes, there are some refactors here. I think the names of the methods that deal with proto resources and files are too similar to the non-proto methods, so I felt moving the "proto" to a prefix was warranted. I have been reconsidering some of these design choices as I backfill tests.

I wanted to have test coverage before changing the implementation to use `ruby_package`, because not every proto service uses it. I had considered submitting separate PRs, but I also don't want to delay the change as Ads is wanting this issue fixed.

Reply:
The difference between `proto_service` and `proto_services` is that the protobuf plugin creates the `proto_service` file, but the grpc plugin creates the `proto_services` file. It looks like the grpc plugin appends `_services` to the `.proto` file. So given a file like `google/cloud/speech/v1/cloud_speech.proto` we would see the following:

* `proto_service_require`: `google/cloud/speech/v1/cloud_speech_pb`
* `proto_services_require`: `google/cloud/speech/v1/cloud_speech_services_pb`

I am open to a different/better naming convention.

Reply:
The reason why this particular method was renamed from `services` to `service` is that `services` only really applies to the file names, and not the Ruby constructs that implement the service.